src/disk_manager.rs

```diff
@@ -411,6 +426,65 @@ fn safe_to_remove_disk(
     Ok(())
 }
 
+ pub fn get_jira_tickets(s: &mut Socket, config_dir: &Path) -> BynarResult<()> {
+    let mut result = OpJiraTicketsResult::new();
+    let config: ConfigSettings = match helpers::load_config(&config_dir, "bynar.json") {
+        Ok(p) => p,
+        Err(e) => {
+            error!("Failed to load config file {}", e);
+            result.set_result(ResultType::ERR);
+            result.set_error_msg(e.to_string());
+
+            // unable to load config file
+            let _ = respond_to_client(&result, s);
+            return Ok(());
+        }
+    };
+    let db_config = config.database;
+    let db_pool = match in_progress::create_db_connection_pool(&db_config) {
+        Ok(p) => p,
+        Err(e) => {
+            error!("Failed to create database pool {}", e);
+            result.set_result(ResultType::ERR);
+            result.set_error_msg(e.to_string());
+
+            // unable to create DB connection
+            let _ = respond_to_client(&result, s);
+            return Ok(());
+        }
+    };
+
+    let h_info = Host::new();
+    if h_info.is_err() {
+        error!("Failed to gather host information");
+        //gracefully exit
+        return Ok(());
+    }
+    let host_info = h_info.expect("Failed to gather host information");
+    debug!("Gathered host info: {:?}", host_info);
+
+    info!("Getting all  outstanding repair tickets");
+    let tickets = in_progress::get_all_pendout_tickets(&db_pool)?;
+    debug!("outstanding tickets: {:?}", tickets);
+    info!("Checking for resolved repair tickets");
+    result.set_result(ResultType::OK);
+    let proto_jira: Vec<JiraInfo> = tickets
+        .iter()
+        .map(|j| {
+            let mut jira_result = JiraInfo::new();
+            jira_result.set_ticket_id(j.ticket_id.clone());
+            jira_result.set_server_name(j.device_name.clone());
```

Comment:
server name is not device name. I think get_all_pending_tickets() should get the storage_detail_id and use that to reverse map that to the server name/hostname to which it belongs. The device name is the name of the device that is being replaced/repaired.